samples/cpp/intelligent_scissors.cpp

```diff
@@ -114,59 +117,64 @@ static void find_min_path(const Point& start)
     }
 }
 
-static void onMouse(int event, int x, int y, int , void *)
+
+static void onMouse(int event, int x, int y, int , void* userdata)
 {
+    Calculated_parameters* tmp = reinterpret_cast<Calculated_parameters*>(userdata);
     if (event == EVENT_LBUTTONDOWN)
     {
-        end = Point(x, y);
-        if (!contours.back().empty())
+        tmp->end = Point(x, y);
+        if (!tmp->contours.back().empty())
         {
-            for (int i = static_cast<int>(tmp_contour.size()) - 1; i >= 0; i--)
+            for (int i = static_cast<int>(tmp->tmp_contour.size()) - 1; i >= 0; i--)
             {
-                contours.back().push_back(tmp_contour[i]);
+                tmp->contours.back().push_back(tmp->tmp_contour[i]);
             }
-            tmp_contour.clear();
+            tmp->tmp_contour.clear();
         }
         else
         {
-            contours.back().push_back(end);
+            tmp->contours.back().push_back(tmp->end);
         }
-        find_min_path(end);
+        find_min_path(tmp->end, tmp);
 
-        img_render.copyTo(img_pre_render);
-        imshow("lasso", img_render);
+        tmp->img_render.copyTo(tmp->img_pre_render);
+        imshow("lasso", tmp->img_render);
     }
     else if (event == EVENT_RBUTTONDOWN)
     {
-        img_pre_render.copyTo(img_render);
-        drawContours(img_pre_render, contours, static_cast<int>(contours.size()) - 1, Scalar(0,255,0), FILLED);
-        addWeighted(img_pre_render, 0.3, img_render, 0.7, 0, img_render);
-        contours.resize(contours.size() + 1);
-        imshow("lasso", img_render);
+        tmp->img_pre_render.copyTo(tmp->img_render);
+        drawContours(tmp->img_pre_render, tmp->contours, static_cast<int>(tmp->contours.size()) - 1, Scalar(0,255,0), FILLED);
+        addWeighted(tmp->img_pre_render, 0.3, tmp->img_render, 0.7, 0, tmp->img_render);
+        tmp->contours.resize(tmp->contours.size() + 1);
+        imshow("lasso", tmp->img_render);
     }
-    else if (event == EVENT_MOUSEMOVE && !contours.back().empty())
+    else if (event == EVENT_MOUSEMOVE && !tmp->contours.back().empty())
     {
-        tmp_contour.clear();
-        img_pre_render.copyTo(img_render);
+        tmp->tmp_contour.clear();
+        tmp->img_pre_render.copyTo(tmp->img_render);
         Point val_point = Point(x, y);
-        while (val_point != end)
+        while (val_point != tmp->end)
         {
-            tmp_contour.push_back(val_point);
-            Point cur = Point(hit_map_x.at<int>(val_point), hit_map_y.at<int>(val_point));
-            line(img_render, val_point, cur, Scalar(255, 0, 0), 2);
+            tmp->tmp_contour.push_back(val_point);
+            Point cur = Point(tmp->hit_map_x.at<int>(val_point), tmp->hit_map_y.at<int>(val_point));
+            line(tmp->img_render, val_point, cur, Scalar(255, 0, 0), 2);
             val_point = cur;
         }
-        imshow("lasso", img_render);
+        imshow("lasso", tmp->img_render);
```

Comment:
there are too much `tmp->` as for me. Can you just get them at the beginning of method?
```cpp
std::vector<Point> tmp_contour = tmp->tmp_contour;
// ...
```